control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -137,6 +138,23 @@ func Setup(opts *operator.HostedClusterConfigOperatorConfig) error {
 		return fmt.Errorf("failed to create uncached client: %w", err)
 	}
 
+	var kubevirtInfraClient client.Client
+	if opts.KubevirtInfraConfig != opts.Config {
+		// if kubevirt infra config is not used, it is being set as the mgmt config
+		kubevirtInfraClient, err = client.New(opts.KubevirtInfraConfig, client.Options{
+			Scheme: opts.Manager.GetScheme(),
+			Mapper: opts.Manager.GetRESTMapper(),
+			Opts: client.WarningHandlerOptions{
+				SuppressWarnings: true,
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to create kubevirt infra uncached client: %w", err)
+		}
```

Comment:
i'd suggest always creating the kubevirtInfraClient rather than trying to reuse a client if we detect that the ```opts.KubevirtInfraConfig``` matches another known client config.

Reply:
code updated.